component/web/oauth-common/src/main/java/org/gatein/security/oauth/openid/OpenIdProcessorImpl.java

```diff
@@ -426,34 +428,76 @@ protected JSONObject parseResponse(String httpResponse) throws JSONException {
 
         return uinfo;
     }
-    
+
     @Override
     public void revokeToken(OpenIdAccessTokenContext accessToken) throws OAuthException {
-    
+
     }
-    
+
     // Parse given String and add all scopes from it to given Set
     private void addScopesFromString(String scope, Set<String> scopes) {
         String[] scopes2 = scope.split(" ");
         for (String current : scopes2) {
             scopes.add(current);
         }
     }
-    
+
     protected URL sendAccessTokenRequest() throws IOException {
         if (log.isTraceEnabled())
             log.trace("AccessToken Request=" + this.accessTokenURL);
         return new URL(this.accessTokenURL);
     }
-    
+
     protected URL getTokenInfoURL() throws IOException {
         if (log.isTraceEnabled())
             log.trace("TokenInfo Request=" + this.tokenInfoURL);
         return new URL(this.tokenInfoURL);
     }
+
     protected URL getUserInfoURL() throws IOException {
         if (log.isTraceEnabled())
             log.trace("UserInfo Request=" + this.userInfoURL);
         return new URL(this.userInfoURL);
     }
-}
+
+    @Override
+    public void start() {
+        if (StringUtils.isBlank(this.wellKnownConfigurationUrl)) {
+            log.error("wellKnownConfigurationUrl is not configured");
+            return;
+        }
+        try {
+            String wellKnownConfigurationContent = readUrl(new URL(this.wellKnownConfigurationUrl));
+            if (wellKnownConfigurationContent != null) {
+                JSONObject json = new JSONObject(wellKnownConfigurationContent);
+                this.authenticationURL = json.getString("authorization_endpoint");
+                this.accessTokenURL = json.getString("token_endpoint");
+                this.userInfoURL = json.getString("userinfo_endpoint");
+            }
+        } catch (JSONException e) {
+            log.error("Unable to read webKnownUrl content.", e.getMessage());
+        } catch (MalformedURLException e) {
+            log.error("Unknown wellKnownConfigurationUrl :" + e.getMessage());
```

Comment:
Message should be something like 
"WellKnowConfigurationUrl malformed : url"
We should display the url in error log in order to be able to see why there is a problem